### PR TITLE
Use `DO_ASSERT()` instead of `assert()` thoughout tread.cpp and twrite.cpp

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**

### CODE REVIEW

File: src/engraving/rw/400/twrite.cpp [hunk 1]
```diff
@@ -198,6 +198,9 @@ class WriteVisitor : public rtti::Visitor<WriteVisitor>
 void TWrite::writeItem(const EngravingItem* item, XmlWriter& xml, WriteContext& ctx)
 {
     bool found = WriteVisitor::visit(WriteTypes {}, item, xml, ctx);
+#ifdef NDEBUG // avoid compiler warning
+    UNUSED(found);
+#endif
     DO_ASSERT(found);
```
Comment (igorkorsukov) [inline src/engraving/rw/400/twrite.cpp:204]: Thanks, but I already fixed it, I improved the `DO_ASSERT` macro, 
 (the fix will be in my next PR)

Comment (igorkorsukov) [reply src/engraving/rw/400/twrite.cpp:204]: https://github.com/musescore/MuseScore/pull/17365/commits/9f33cede7c3a4da5f4ff7fc1c2440c895e57266c

Comment (igorkorsukov) [reply src/engraving/rw/400/twrite.cpp:204]: > (Wouldn't it be wise to have that UNUSED() inside these Q_ASSERT() macros?)

Exactly!

Comment (Jojo-Schmitz) [reply src/engraving/rw/400/twrite.cpp:204]: I tried the same, but got tons of compiler errors (when `cond` is something like `value = function()`)

Comment (igorkorsukov) [reply src/engraving/rw/400/twrite.cpp:204]: I think `value = function()` should not be, maybe just `function()`

Comment (Jojo-Schmitz) [reply src/engraving/rw/400/twrite.cpp:204]: Ah, it is gone, guess I didn't do a `make clean` before.

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**